File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.cpp
class Solution {
public:
        string removeOuterParentheses(string S) {
        string ans;
        int opened = 0;
        for(int i=0;i<S.size();i++){
            if(S[i]=='('){
                if(opened>0) ans+=S[i];
                opened++;
            }
            if(S[i]==')'){
                opened--;
                if(opened>0) ans+=S[i];
            }
        }
        return ans;
    }
};

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String S) {
        StringBuilder ans = new StringBuilder();
        int opened = 0;

        for(int i = 0; i < S.length(); i++) {

            if(S.charAt(i) == '(') {
                // Keep '(' only if it is not the outermost one
                if(opened > 0)
                    ans.append(S.charAt(i));

                opened++;
            }

            if(S.charAt(i) == ')') {
                opened--;

                // Keep ')' only if it is not the outermost one
                if(opened > 0)
                    ans.append(S.charAt(i));
            }
        }

        return ans.toString();
    }
}